packages/effect/src/Array.ts

```diff
@@ -1055,8 +1055,17 @@ export const insertAt: {
  * @since 2.0.0
  */
 export const replace: {
-  <B>(i: number, b: B): <A>(self: Iterable<A>) => Array<A | B>
-  <A, B>(self: Iterable<A>, i: number, b: B): Array<A | B>
+  <B>(
+    i: number,
+    b: B
+  ): <A, S extends Iterable<A> = Iterable<A>>(
```

Comment:
what's the purpose of these defaults? (`= Iterable<A>`)

Reply:
keep backward compatibility in case of use with passing type arguments
in the previous version, there was only one required parameter